### PR TITLE
Parts for small support vehicle ammo tracking

### PR DESCRIPTION
**Pull request by neoancient**
Adds InfantryAmmoBin and InfantryAmmoStorage for handling small support vehicle ammo repairs and reloading. Each weapon has at least one ammo bin. If it can use inferno ammo it has two. There is also a dialog for redistributing space between standard and inferno ammo.

Depends on MegaMek/megamek#2219.

### CODE REVIEW

**Review by NickAragua — COMMENTED** [review]
One question, looks good otherwise.

File: MekHQ/data/mechfiles/vehicles/TRO Vehicle Annex/Tracked/RRV.blk [hunk 1]
```diff
@@ -38,8 +38,8 @@ Tracked
 
 <transporters>
 cargobay:1.054:1:1
-troopspace:1.0
 standardseats:3.0:0
+troopspace:1.0
```
[the remark this answers is not in the document]
Comment (neoancient) [reply MekHQ/data/mechfiles/vehicles/TRO Vehicle Annex/Tracked/RRV.blk:42]: No, but I copied it over for testing and accidentally committed it. I noticed a couple commits later and didn't think it made enough of a difference to go back and revert it.

**Review by NickAragua — APPROVED** [review]

File: MekHQ/src/mekhq/gui/dialog/SmallSVAmmoSwapDialog.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2017 - The MegaMek Team
```
Comment (Windchild292) [inline MekHQ/src/mekhq/gui/dialog/SmallSVAmmoSwapDialog.java:2]: Copyright 2017 on a new dialog?

File: MekHQ/src/mekhq/gui/dialog/SmallSVAmmoSwapDialog.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (Windchild292) [inline MekHQ/src/mekhq/gui/dialog/SmallSVAmmoSwapDialog.java:17]:
```suggestion
 * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
```

File: MekHQ/src/mekhq/gui/dialog/SmallSVAmmoSwapDialog.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
```
Comment (Windchild292) [inline MekHQ/src/mekhq/gui/dialog/SmallSVAmmoSwapDialog.java:13]:
```suggestion
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
```

Comment (Windchild292) [reply MekHQ/src/mekhq/gui/dialog/SmallSVAmmoSwapDialog.java:13]: Double space is a potential dyslexia issue

File: MekHQ/src/mekhq/campaign/parts/equipment/MissingInfantryAmmoBin.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/parts/equipment/MissingInfantryAmmoBin.java:17]:
```suggestion
 * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
```

File: MekHQ/src/mekhq/campaign/parts/equipment/MissingInfantryAmmoBin.java [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/parts/equipment/MissingInfantryAmmoBin.java:13]:
```suggestion
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
```

Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/parts/equipment/MissingInfantryAmmoBin.java:13]: Double space is a potential dyslexia issue

File: MekHQ/src/mekhq/campaign/parts/equipment/InfantryAmmoBin.java [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/parts/equipment/InfantryAmmoBin.java:13]:
```suggestion
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
```

File: MekHQ/src/mekhq/campaign/parts/equipment/InfantryAmmoBin.java [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/parts/equipment/InfantryAmmoBin.java:17]:
```suggestion
 * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
```

File: MekHQ/src/mekhq/campaign/parts/InfantryAmmoStorage.java [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/parts/InfantryAmmoStorage.java:13]:
```suggestion
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
```

File: MekHQ/src/mekhq/campaign/parts/InfantryAmmoStorage.java [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/parts/InfantryAmmoStorage.java:17]:
```suggestion
 * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
```

File: MekHQ/src/mekhq/campaign/parts/InfantryAmmoStorage.java (hunk 9 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/parts/InfantryAmmoStorage.java:13]: Double space is a potential dyslexia issue

**Review by Windchild292 — APPROVED** [review]
Just a few code formatting fixes for dyslexia issues.

**Pull request merged**